services/shhext/api.go

```diff
@@ -280,6 +280,42 @@ func (api *PublicAPI) SendDirectMessage(ctx context.Context, msg chat.SendDirect
 	return response, nil
 }
 
+// SendPairingMessage sends a 1:1 chat message to our own devices to initiate a pairing session
+func (api *PublicAPI) SendPairingMessage(ctx context.Context, msg chat.SendDirectMessageRPC) ([]hexutil.Bytes, error) {
+	if !api.service.pfsEnabled {
+		return nil, ErrPFSNotEnabled
+	}
+	// To be completely agnostic from whisper we should not be using whisper to store the key
+	privateKey, err := api.service.w.GetPrivateKey(msg.Sig)
```

Comment:
`Sig` in our version of whisper is the public key of the sender, (it would be a key id in normal whisper), but it's the same code as whisper https://github.com/ethereum/go-ethereum/blob/f951e23fb5ad2f7017f314a95287bc0506a67d05/whisper/whisperv6/api.go#L255

Reply:
Oh because we have that determinism implemented. Remember now :)